ffi/capi/src/properties_unisets.rs

```diff
@@ -65,5 +65,23 @@ pub mod ffi {
                 )?,
             )))
         }
+
+        /// Get the `Basic_Emoji` value for a given character, using compiled data
+        #[diplomat::rust_link(icu::properties::props::EmojiSet::for_char, FnInTrait)]
+        #[cfg(feature = "compiled_data")]
+        pub fn basic_emoji_for_char(ch: DiplomatChar) -> bool {
+            icu_properties::EmojiSetData::new::<BasicEmoji>().contains32(ch)
+        }
+
+        /// Get the `Basic_Emoji` value for a given character, using compiled data
+        #[diplomat::rust_link(icu::properties::props::EmojiSet::for_str, FnInTrait)]
+        #[cfg(feature = "compiled_data")]
+        pub fn basic_emoji_for_str(s: &DiplomatStr) -> bool {
+            use icu_properties::props::EmojiSet;
+            let Ok(s) = core::str::from_utf8(s) else {
+                return false;
+            };
+            BasicEmoji::for_str(s)
```

Comment:
observation: this does a full UTF-8 validation, which is unnecessary for the lookup. I see that the non-convenience API *also* does this, we should improve that by adding `contains_utf8` to the `EmojiSet` type.

it might make sense to define the FFI convenience API in terms of the FFI API, not in terms of the Rust convenience API. This way, we only have one code location where we do the UTF-8 validation, and won't forget to fix this if we fix the other location

Reply:
> it might make sense to define the FFI convenience API in terms of the FFI API

That forces an allocation, one of the reasons that FFI convenience APIs are necessary (arguably, FFI convenience APIs are more necessary than Rust ones).

But we can define the convenience API in terms of the regular Rust API instead of the convenience API, to solve your concern.

Reply:
yes sorry that's what I meant